flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSinkITCase.java

```diff
@@ -576,6 +576,24 @@ public void testWriteSuccessFile() throws Exception {
         assertThat(new File(changeFileNameTablePath, "dt=2022-08-15/_ZM")).exists();
     }
 
+    @Test
+    public void testInsertBothDynamicAndStaticPartitionWithoutData() throws Exception {
```

Comment:
I think we can just add the test case in HiveTableSinkITCase#testWritingNoDataToPartition.